external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/SerializableCredentialsProvider.scala

```diff
@@ -83,3 +84,146 @@ private[kinesis] final case class STSCredentialsProvider(
     }
   }
 }
+
+@InterfaceStability.Stable
+object SerializableCredentialsProvider {
```

Comment:
I agree we should definitely come up with a better name here. What about ```SparkAWSCredentials```? Obviously it's not as succinct as ```AWSCredentials``` but I think it's a clear name that avoids collisions.

I'm okay with ```CredentialsProvider``` otherwise.